extension/types/unicode-constants.ts

```diff
@@ -0,0 +1,17 @@
+export enum KANA {
+  HIRAGANA_START = 0x3041,
+  HIRAGANA_END = 0x309f,
+  KATAKANA_START = 0x30a0,
+  KATAKANA_END = 0x30ff,
+  HW_KATAKANA_START = 0xff66,
+  HW_KATAKANA_END = 0xff9d,
+}
+
+export enum PUNCTUATION {
```

Comment:
Since these are used to check equality is there any reason to use their code points rather than strings?

Reply:
We can change them to the actual strings, however, I thought for consistency we could keep them all as code points. I guess it is just a bit of OCD on my end. 

Having all code points and then one :
`ZERO_WIDTH_JOINER = '‌',` <- The character is actually inside of the quotes.

Also I thought this would be slightly confusing to look at.
```
console.log("～".charCodeAt(0));
console.log("〜".charCodeAt(0));
```
from first glance they appear to be the same but they are actually different.
```
65374 or 0xff5e
12316 or 0x301c
```

If you think otherwise, I can just swap it back to having the characters themselves.

Reply:
I guess both ways can make it hard to tell at a glance what the character is but at least with the unicode style, it's unambiguous in that you know how to look it up.

I think I was just thinking it seemed weird to convert a string to a code point when we controlled the lookup table.

One thing I'd recommend (if we're not already doing this) is add a comment with the string character next to/above the unicode symbols in case someone wants to copy it quickly. Might require some clever thinking to make it obvious in the case of invisible characters. :)